experimental/AlgebraicStatistics/src/GaussianGraphicalModels.jl

```diff
@@ -1,217 +1,283 @@
-  S, s = polynomial_ring(K, varnames; cached=cached)
-  d = Dict([varindices[i] => s[i] for i in 1:length(varindices)])
-  cov_matrix = matrix([[i < j ? d[i,j] : d[j,i] for j in 1:n] for i in 1:n])
-  GaussianRing(S, d, cov_matrix)
-end
+julia> G = graph_from_edges(Directed, [[1, 2], [2, 3]])
+Directed graph with 3 nodes and the following edges:
+(1, 2)(2, 3)
 
-function Base.show(io::IO, R::GaussianRing)
-  coeffs = base_ring(R.ring)
-  k = ngens(R.ring)
-  print(io, "Gaussian ring over $(coeffs) in $(k) variables", "\n", chop(string(gens(R.ring)), head = 16, tail = 1))
-end
+julia> GM = gaussian_graphical_model(G)
+Gaussian Graphical Model on a Directed graph with 3 nodes and 2 edges
 
-# functions for getting attributes of Gaussian rings
+julia> typeof(GM)
+GaussianGraphicalModel{Graph{Directed}, Nothing}
 
-@doc raw"""
-    ring(R::GaussianRing)
-
-Return the multivariate polynomial ring inside `R`.
-
-## Examples
+julia> CG = graph_from_labeled_edges(Dict((1, 2) => "red", (2, 3) => "green"); name=:color)
+Undirected graph with 3 nodes and the following labeling(s):
+label: color
+(2, 1) -> red
+(3, 2) -> green
 
-```jldoctest
-julia> R = gaussian_ring(3)
-Gaussian ring over Rational field in 6 variables
-s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
-
-julia> ring(R)
-Multivariate polynomial ring in 6 variables s[1, 1], s[1, 2], s[1, 3], s[2, 2], ..., s[3, 3]
-  over rational field
+julia> GM = gaussian_graphical_model(CG)
+Gaussian Graphical Model on a Undirected graph with 3 nodes and 2 edges with labeling(s) [:color]
 ```
 """
-function ring(R::GaussianRing)
-  R.ring
+function gaussian_graphical_model(G::Graph{T}; s_varname::VarName="s", l_varname::VarName="l", w_varname::VarName="w") where T <: Directed
+  @req is_acyclic(G) "$G must be acyclic"
+  GaussianGraphicalModel(G, Dict{Symbol, VarName}(:s => s_varname, :l => l_varname, :w => w_varname))
 end
 
-@doc raw"""
-    gens(R::GaussianRing)
+function gaussian_graphical_model(G::Graph{Undirected}; s_varname::VarName="s", k_varname::VarName="k")
+  GaussianGraphicalModel(G, Dict{Symbol, VarName}(:s => s_varname, :k => k_varname))
+end
 
-Return the generators of the multivariate polynomial ring inside the GaussianRing as a dictionary which can be easily indexed
+function gaussian_graphical_model(G::MixedGraph; s_varname::VarName="s", l_varname::VarName="l", w_varname::VarName="w")
+  GaussianGraphicalModel(G, Dict{Symbol, VarName}(:s => s_varname, :l => l_varname, :w => w_varname))
+end
 
-## Examples
+varnames(GM::GaussianGraphicalModel) = GM.varnames
 
-```jldoctest; filter = Main.Oscar.doctestfilter_hash_changes_in_1_13()
-julia> R = gaussian_ring(3)
-Gaussian ring over Rational field in 6 variables
-s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
+function Base.show(io::IO, M::GaussianGraphicalModel{T, L}) where {T, L}
+  io = pretty(io)
+  if is_terse(io)
+    print(io, "Gaussian Graphical Model on a")
+    !(L == Nothing) && print(io, " labelled")
```

Comment:
Comparison to `nothing` or to types should ideally use `===` or `!===` to avoid risk of invalidations
```suggestion
    !(L === Nothing) && print(io, " labelled")
```
or
```suggestion
    (L !== Nothing) && print(io, " labelled")
```